node_modules/gh-orgunit/lib/api.js

```diff
@@ -557,51 +563,90 @@ var addOrgUnitSeries = module.exports.addOrgUnitSeries = function(ctx, id, serie
-                return !_.find(orgUnit.Series, function(orgUnitSerie) {
-                    return (orgUnitSerie.id === serie.id);
+                // Ensure the series are from the same application as the organisational unit
+                var otherAppSeries = _.filter(series, function(serie) {
+                    return (orgUnit.AppId !== serie.AppId);
                 });
-            });
+                if (!_.isEmpty(otherAppSeries)) {
+                    return callback({'code': 400, 'msg': 'Some of the provided series belong to another application'});
+                }
 
-            // If there are no new series, we can return early
-            if (_.isEmpty(newSeries)) {
-                return callback();
-            }
+                // Get all the series that are NOT part of the organisational unit already
+                var newSeries = _.filter(series, function(serie) {
+                    return !_.find(orgUnit.Series, function(orgUnitSerie) {
+                        return (orgUnitSerie.id === serie.id);
+                    });
+                });
 
-            // Ensure the current user can add series to the organisational unit
-            OrgUnitAuthz.canAddOrgUnitSeries(ctx, orgUnit, function(err, canAdd) {
-                if (err) {
-                    return callback(err);
-                } else if (!canAdd) {
-                    log().warn({
-                        'actor': ctx.user.id,
-                        'displayName': orgUnit.displayName
-                    }, 'An unauthorized attempt at adding series to an organisational unit occurred');
-                    return callback({'code': 401, 'msg': 'You are not allowed to add series to the organisational unit'});
+                // If there are no new series, we can return early
+                if (_.isEmpty(newSeries)) {
+                    return callback();
                 }
 
-                OrgUnitDAO.addOrgUnitSeries(orgUnit, series, callback);
+                // Ensure the current user can add series to the organisational unit
+                OrgUnitAuthz.canAddOrgUnitSeries(ctx, orgUnit, function(err, canAdd) {
+                    if (err) {
+                        return callback(err);
+                    } else if (!canAdd) {
+                        log().warn({
+                            'actor': ctx.user.id,
+                            'displayName': orgUnit.displayName
+                        }, 'An unauthorized attempt at adding series to an organisational unit occurred');
+                        return callback({'code': 401, 'msg': 'You are not allowed to add series to the organisational unit'});
+                    }
+
+                    OrgUnitDAO.addOrgUnitSeries(orgUnit, series, fromOrgUnit, callback);
+                });
             });
         });
     });
 };
 
 /**
+ * Validate that an id points to a valid organisational unit belonging to the correct application.
```

Comment:
`Ensure that an organisational belongs to the correct application and verify that a set of event series belong to the organisation unit. If no organisation unit is provided, no validation will happen`